### PR TITLE
XX Fix issue in the visit_Subroutine

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/lfortran/semantics/ast_symboltable_visitor.cpp [hunk 1]
```diff
@@ -500,11 +500,10 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
...
         if (parent_scope->get_symbol(sym_name) != nullptr) {
             ASR::symbol_t *f1 = parent_scope->get_symbol(sym_name);
-            ASR::Function_t *f2 = nullptr;
-            if( ASR::is_a<ASR::Function_t>(*f1) ) {
-                f2 = ASR::down_cast<ASR::Function_t>(f1);
-            }
+            ASR::Function_t *f2 = ASR::down_cast<ASR::Function_t>(
+                ASRUtils::symbol_get_past_external(f1));
             if ((f1->type == ASR::symbolType::ExternalSymbol && in_submodule) ||
+                f2->m_access == ASR::accessType::Private ||
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/semantics/ast_symboltable_visitor.cpp:506]: I feel this is not the correct fix!

**Comment by Thirumalai-Shaktivel** [comment]
I have a doubt:
https://github.com/lfortran/lfortran/blob/c8510351d19cb289b179cffed0fb8a566086037b/src/lfortran/semantics/ast_symboltable_visitor.cpp#L502-L513
Why do we erase a symbol here? 
How does this check whether Subroutine is already defined or not?
@certik @czgdp1807

**Comment by certik** [comment]
I am not sure. What you can do is remove the line 510 and run all tests and see if anything breaks.

**Comment by Thirumalai-Shaktivel** [comment]
This issue is fixed in, so closing this.

**Pull request closed without merging**